sdk/storage_datalake/src/operations/file_systems_list.rs

```diff
@@ -0,0 +1,129 @@
+use crate::clients::DataLakeClient;
+use crate::file_system::{FileSystem, FileSystemList};
+use azure_core::AppendToUrlQuery;
+use azure_core::{collect_pinned_stream, prelude::*, Pageable, Response};
+use azure_storage::core::headers::CommonStorageResponseHeaders;
+use std::convert::TryInto;
+use std::pin::Pin;
+
+#[derive(Debug, Clone)]
+pub struct ListFileSystems {
+    client: DataLakeClient,
+    prefix: Option<Prefix>,
+    next_marker: Option<NextMarker>,
+    max_results: Option<MaxResults>,
+    client_request_id: Option<ClientRequestId>,
+    timeout: Option<Timeout>,
+    context: Option<Context>,
+}
+
+impl ListFileSystems {
+    pub(crate) fn new(client: DataLakeClient, context: Option<Context>) -> Self {
+        Self {
+            client,
+            prefix: None,
+            next_marker: None,
+            max_results: None,
+            client_request_id: None,
+            timeout: None,
+            context,
+        }
+    }
+
+    setters! {
+        prefix: Prefix => Some(prefix),
+        next_marker: NextMarker => Some(next_marker),
+        max_results: MaxResults => Some(max_results),
+        client_request_id: ClientRequestId => Some(client_request_id),
+        timeout: Timeout => Some(timeout),
+        context: Context => Some(context),
+    }
+
+    pub fn into_stream(self) -> Pin<Box<Pageable<ListFileSystemsResponse>>> {
```

Comment:
I wish we didn't have to return a `Pin<Box<T>>` 🙁. The reason we haven't returned `Pageable` wrapped in a `Pin<Box<T>>` is because that forces the user into heap pinning instead of allowing them to pin on the stack. However, it's probably not worth the ergonomics hit so returning a `Box<Pin<T>>` is fine.

Reply:
I agree :).